tests/unit/test_req.py

```diff
@@ -636,3 +637,25 @@ def test_mismatched_versions(caplog, tmpdir):
         'Requested simplewheel==2.0, '
         'but installing version 1.0'
     )
+
+
+class TestRequirementTracker(object):
+    @pytest.fixture(autouse=True)
+    def blank_env(self):
+        with patch.dict(os.environ, clear=True):
+            yield
+
+    def test_no_side_effects_on_init(self):
+        RequirementTracker()
+        assert 'PIP_REQ_TRACKER' not in os.environ
+
+    def test_keeps_original_environ(self, tmpdir):
+        os.environ['PIP_REQ_TRACKER'] = str(tmpdir)
+        with RequirementTracker():
+            assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+        assert os.environ['PIP_REQ_TRACKER'] == str(tmpdir)
+
+    def test_undoes_environ_set(self):
+        with RequirementTracker():
```

Comment:
I would probably also `assert 'PIP_REQ_TRACKER' not in os.environ` before starting the manager just to be safe and so it's easier to see by looking at the test that it's working (even though not strictly necessary).

Reply:
there's an autouse fixture which clears the environ -- seems unnecessary to assert imo